sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -172,7 +172,12 @@ protected Symbol convertFunctionCall(FunctionCall node, ExpressionAnalysisContex
                 ImmutableList.<DataType>of(new SetType(argumentTypes.get(0)));
 
             FunctionIdent ident = new FunctionIdent(nodeName, outerArgumentTypes);
-            functionInfo = getFunctionInfo(ident);
+            try {
+                functionInfo = getFunctionInfo(ident);
+            } catch (UnsupportedOperationException ex) {
+                throw new UnsupportedOperationException("Function " + node.getName().toString() + "() " +
+                    "is not supported in combination with distinct", ex);
```

Comment:
@seut any suggestions on this? In our oppinion `unknown function collection_max(long_set)` is also not really helpful